webactions.md

```diff
@@ -0,0 +1,546 @@
+https://${APIHOST}/api/v1/web/guest/demo/hello
+```
+
+**[[NH: check link to webactions.md.]]**
+To configure a web action with credentials, refer to [Securing web actions](webactions.md#securing-web-actions).
+
+### The URL for a web action
+
+A web action can be invoked using a URL that is structured as follows:
+`https://{APIHOST}/api/v1/web/{QUALIFIED ACTION NAME}.{EXT}`.
+
+The fully qualified name of an action consists of three parts: the namespace, the package name, and the action name. In our example, it's `my-namespace/demo/hello`.
+
+**Note:** If the action is not in a named package, use the package name  `default`. For example, if the `hello` action is not in a named package, use the URL `my-namespace/default/hello`.
+
+The last part of the URI is called the `extension`. It's typically `.http`, although other values are permitted, as described later.
+
+The web action API path can be used with `curl` or `wget` without an API key. It can even be entered directly in your browser.
+
+Try opening the following URL in your web browser, substituting the name of your Nimbella namespace for `my-namespace`.
+
+```
+[https://${APIHOST}/api/v1/web/my-namespace/demo/hello.http?name=Jane](https://${APIHOST}/api/v1/web/my-namespace/demo/hello.http?name=Jane).
+```
+
+Or try invoking the action via `curl`, again substituting the name of your Nimbella namespace for `my-namespace`:
+
+```
+$ curl https://${APIHOST}/api/v1/web/my-namespace/demo/hello.http?name=Jane
+```
+
+### Other web action examples
+
+Here's an example of a web action that performs an HTTP redirect:
+```javascript
+function main() {
+  return {
+    headers: { location: 'http://nimbella.io' },
+    statusCode: 302
+  }
+}
+```
+
+Or sets a cookie:
+```javascript
+function main() {
+  return {
+    headers: {
+      'Set-Cookie': 'UserID=Jane; Max-Age=3600; Version=',
+      'Content-Type': 'text/html'
+    },
+    statusCode: 200,
+    body: '<html><body><h3>hello</h3></body></html>' }
+}
+```
+
+Or sets multiple cookies:
+```javascript
+function main() {
+  return {
+    headers: {
+      'Set-Cookie': [
+        'UserID=Jane; Max-Age=3600; Version=',
+        'SessionID=asdfgh123456; Path = /'
+      ],
+      'Content-Type': 'text/html'
+    },
+    statusCode: 200,
+    body: '<html><body><h3>hello</h3></body></html>' }
+}
+```
+
+Or returns an `image/png`:
+```javascript
+function main() {
+    let png = <base 64 encoded string>
+    return { headers: { 'Content-Type': 'image/png' },
+             statusCode: 200,
+             body: png };
+}
+```
+
+Or returns `application/json`:
+```javascript
+function main(params) {
+    return {
+        statusCode: 200,
+        headers: { 'Content-Type': 'application/json' },
+        body: params
+    };
+}
+```
+
+The default `content-type` for an HTTP response is `application/json` and the body can be any allowed JSON value. The default `content-type` can be omitted from the headers.
+
+**[[NH: Note link to reference.md]]**
```

Comment:
I'm asking @rabbah to comment on which linked documents should be in scope for your translation effort versus which should be dropped.  He knows what's in the other documents and whether or not translation is worth it.

Reply:
we will need our own reference.md to document the limits - starting with the apache version is ok.

Reply:
ok modified the sentence a bit so people are not surprised when they click the link.